components/dash-core-components/src/components/Input.react.js

```diff
@@ -140,14 +162,22 @@ export default class Input extends PureComponent {
             });
             this.input.current.checkValidity();
         }
-        return this.props.debounce && e.key === 'Enter' && this.onEvent();
+        return (
+            this.props.debounce === true && e.key === 'Enter' && this.onEvent()
+        );
     }
 
     onChange() {
-        if (!this.props.debounce) {
+        const {debounce} = this.props;
+        if (debounce) {
+            if (Number.isFinite(debounce)) {
+                this.debounceEvent(debounce);
+            }
+            if (this.props.type !== 'number') {
+                this.setState({value: this.input.current.value});
+            }
+        } else {
             this.onEvent();
-        } else if (this.props.type !== 'number') {
-            this.setState({value: this.input.current.value});
```

Comment:
Was this moved somewhere else?

Reply:
Nope, this is still here. Since I added another condition to the "if/else", I just inverted the logic to keep it readable.